File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/AttIdValue.java
package ru.citeck.ecos.records3.record.op.atts.service.value.impl;

import ru.citeck.ecos.commons.utils.MandatoryParam;
import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue;

public class AttIdValue implements AttValue {

    private final String id;

    public AttIdValue(Object id) {
        MandatoryParam.check("id", id);
        this.id = id.toString();
    }

    @Override
    public String asText() {
        return id;
    }

    @Override
    public String getId() {
        return id;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/AttIdValue.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.impl

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

class AttIdValue(private val id: Any?) : AttValue {

    override fun asText(): String? {
        return id?.toString() ?: ""
    }

    override fun getId(): Any {
        return id ?: ""
    }
}
